src/test/unit/EigenPodManagerUnit.t.sol

```diff
@@ -117,220 +85,438 @@ contract EigenPodManagerUnitTests is Test, EigenPodPausingConstants {
-                    )
-                )
-            )
-        );
+    function test_addShares_revert_podOwnerZeroAddress() public {
+        cheats.prank(address(delegationManagerMock));
+        cheats.expectRevert("EigenPodManager.addShares: podOwner cannot be zero address");
+        eigenPodManager.addShares(address(0), 0);
+    }
+
+    function testFuzz_addShares_revert_sharesNegative(int256 shares) public {
+        cheats.assume(shares < 0);
+        cheats.prank(address(delegationManagerMock));
+        cheats.expectRevert("EigenPodManager.addShares: shares cannot be negative");
+        eigenPodManager.addShares(defaultStaker, uint256(shares));
+    }
+
+    function testFuzz_addShares_revert_sharesNotWholeGwei(uint256 shares) public {
+        cheats.assume(int256(shares) >= 0);
+        cheats.assume(shares % GWEI_TO_WEI != 0);
+        cheats.prank(address(delegationManagerMock));
+        cheats.expectRevert("EigenPodManager.addShares: shares must be a whole Gwei amount");
+        eigenPodManager.addShares(defaultStaker, shares);
+    }
+
+    function testFuzz_addShares(uint256 shares) public {
+        // Fuzz inputs
+        cheats.assume(defaultStaker != address(0));
+        cheats.assume(shares % GWEI_TO_WEI == 0);
+        cheats.assume(int256(shares) >= 0);
+
+        // Add shares
+        cheats.prank(address(delegationManagerMock));
+        eigenPodManager.addShares(defaultStaker, shares);
+
+        // Check storage update
+        assertEq(eigenPodManager.podOwnerShares(defaultStaker), int256(shares));
+    }
+
+    /*******************************************************************************
+                                Remove Shares Tests
+    ******************************************************************************/
+
+    function testFuzz_removeShares_revert_notDelegationManager(address notDelegationManager) public {
+        cheats.assume(notDelegationManager != address(delegationManagerMock));
+        cheats.prank(notDelegationManager);
+        cheats.expectRevert("EigenPodManager.onlyDelegationManager: not the DelegationManager");
+        eigenPodManager.removeShares(defaultStaker, 0);
+    }
+
+    function testFuzz_removeShares_revert_sharesNegative(int256 shares) public {
+        cheats.assume(shares < 0);
+        cheats.prank(address(delegationManagerMock));
+        cheats.expectRevert("EigenPodManager.removeShares: shares cannot be negative");
+        eigenPodManager.removeShares(defaultStaker, uint256(shares));
+    }
+    
+    function testFuzz_removeShares_revert_sharesNotWholeGwei(uint256 shares) public {
+        cheats.assume(int256(shares) >= 0);
+        cheats.assume(shares % GWEI_TO_WEI != 0);
+        cheats.prank(address(delegationManagerMock));
+        cheats.expectRevert("EigenPodManager.removeShares: shares must be a whole Gwei amount");
+        eigenPodManager.removeShares(defaultStaker, shares);
+    }
+
+    // Fuzzer rejects too many inputs, unit testsing
```

Comment:
you can start with a small uint size fuzzed input and then multiply it by GWEI_TO_WEI ?
like `uint96 fuzzedInput ... uint256 sharesToRemove = fuzzedInput * GWEI_TO_WEI`

Reply:
Fixed